pkg/schemadsl/compiler/compiler_test.go

```diff
@@ -137,6 +137,58 @@ func TestCompile(t *testing.T) {
 				),
 			},
 		},
+		{
+			"relation with required caveat",
+			&someTenant,
+			`definition simple {
+				relation viewer: user with somecaveat
+			}`,
+			"",
+			[]*core.NamespaceDefinition{
+				namespace.Namespace("sometenant/simple",
+					namespace.Relation("viewer", nil,
+						namespace.AllowedRelationWithCaveat("sometenant/user", "...",
+							namespace.Caveat("somecaveat")),
```

Comment:
why isn't the caveat also parsed with a tenant prefix? TODO for a follow up?

Reply:
Yes, my followup PR handles it